### ISSUE TITLE
5.8.0: error in "MIME::QuotedPrint"

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5908]
Migrated from [rt.perl.org#17107](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=17107) (status was 'resolved')

Searchable as RT17107$

**Comment by p5pRT** [comment]
### From Ulrich.Windl@rz.uni-regensburg.de

#### Created by Ulrich.windl@rz.uni-regensburg.de

This is a bug report for perl from Ulrich\.windl@&#8203;rz\.uni\-regensburg\.de\,
generated with the help of perlbug 1\.34 running under perl v5\.8\.0\.


\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
The "perldelta" manual says&#8203;:
&nbsp;       ·   "MIME&#8203;::QuotedPrint"\, by Gisle Aas\, allows you to
&nbsp;           encode data in quoted\-printable encoding\, as defined
&nbsp;           in RFC 2045 \- MIME \(Multipurpose Internet Mail Exten­
&nbsp;           sions\)\.

&nbsp;               use MIME&#8203;::QuotedPrint;

&nbsp;               $encoded = encode\_qp\("Smiley in Unicode&#8203;: \\x\{263a\}"\);
&nbsp;               $decoded = decode\_qp\($encoded\);

&nbsp;               print $encoded\, "\\n"; \# "Smiley in Unicode&#8203;: =263A"

&nbsp;           See also PerlIO&#8203;::via&#8203;::QuotedPrint\.

However I think that the encoding "=263A" is not correct\. Shouldn't it be
"=26=3A"? Perl says&#8203;:

main&#8203;::\(\-e&#8203;:1\)&#8203;:   1
&nbsp;  DB\<1> use MIME&#8203;::QuotedPrint;

&nbsp;  DB\<2> $encoded = encode\_qp\("Smiley in Unicode&#8203;: \\x\{263a\}"\);

&nbsp;  DB\<3> $decoded = decode\_qp\($encoded\);

&nbsp;  DB\<4> x $decoded
0  'Smiley in Unicode&#8203;: &3A'

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=medium

This perlbug was built using Perl v5.8.0 - Mon Sep  2 18:56:06 UTC 2002
It is being executed now by  Perl v5.8.0 - Mon Sep  2 18:43:25 UTC 2002.

Site configuration information for perl v5.8.0:

Configured by root at Mon Sep  2 18:43:25 UTC 2002.

Summary of my perl5 (revision 5.0 version 8 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.4.19, archname=i586-linux-thread-multi
    uname='linux you 2.4.19 #1 smp mon dec 17 18:25:06 gmt 2001 i686 unknown '
    config_args='-ds -e -Dprefix=/usr -Dusethreads -Di_db -Di_dbm -Di_ndbm -Di_gdbm -Duseshrplib=true'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O3 --pipe',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing'
    ccversion='', gccversion='3.2', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =''
    libpth=/lib /usr/lib /usr/local/lib
    libs=-lnsl -ldl -lm -lpthread -lc -lcrypt -lutil
    perllibs=-lnsl -ldl -lm -lpthread -lc -lcrypt -lutil
    libc=, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.2.5'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic -Wl,-rpath,/usr/lib/perl5/5.8.0/i586-linux-thread-multi/CORE'
    cccdlflags='-fPIC', lddlflags='-shared'

Locally applied patches:



@INC for perl v5.8.0:
    /usr/lib/perl5/5.8.0/i586-linux-thread-multi
    /usr/lib/perl5/5.8.0
    /usr/lib/perl5/site_perl/5.8.0/i586-linux-thread-multi
    /usr/lib/perl5/site_perl/5.8.0
    /usr/lib/perl5/site_perl
    .


Environment for perl v5.8.0:
    HOME=/root
    LANG (unset)
    LANGUAGE (unset)
    LC_CTYPE=de_DE@euro
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)

PATH=/sbin:/usr/sbin:/usr/local/sbin:/root/bin:/usr/local/bin:/usr/bin:/usr/X11R
6/bin:/bin:/usr/games:/opt/gnome2/bin:/opt/gnome/bin:/opt/gnome/bin
    PERL_BADLANG (unset)
    SHELL=/bin/bash


```
</details>

**Comment by p5pRT** [comment]
### From @gisle

"Ulrich Windl" \(via RT\) \<perlbug@&#8203;perl\.org> writes&#8203;:

> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> The "perldelta" manual says&#8203;:
>        ·   "MIME&#8203;::QuotedPrint"\, by Gisle Aas\, allows you to
>            encode data in quoted\-printable encoding\, as defined
>            in RFC 2045 \- MIME \(Multipurpose Internet Mail Exten­
>            sions\)\.
> 
>                use MIME&#8203;::QuotedPrint;
> 
>                $encoded = encode\_qp\("Smiley in Unicode&#8203;: \\x\{263a\}"\);
>                $decoded = decode\_qp\($encoded\);
> 
>                print $encoded\, "\\n"; \# "Smiley in Unicode&#8203;: =263A"
> 
>            See also PerlIO&#8203;::via&#8203;::QuotedPrint\.
> 
> However I think that the encoding "=263A" is not correct\. Shouldn't it be
> "=26=3A"? Perl says&#8203;:

This example is bogus\.  Quoted\-Printable is only defined for bytes\.
The encode\_qp\(\) is supposed to croak on input like this\.  It looks
like this section of perldelta has already been fixed for bleadperl\.

Regards\,
Gisle

**Comment by p5pRT** [comment]
### From @jhi

This documentation error \(MIME&#8203;::QuotedPrint is only for bytes\, not Unicode characters\) has been fixed by change \#17868\, which means that
the fix will be in Perl 5\.8\.1 \(in a new 'perl58delta' file which is essentially the 'perldelta' of 5\.8\.0\, while 'perldelta' will be the
5\.8\.1 changes\)\.  I'm marking the ticket as resolved\.

**Comment by p5pRT** [comment]
@jhi - Status changed from 'new' to 'resolved'
